db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ColumnReference.java

```diff
@@ -985,6 +985,10 @@ public ValueNode remapColumnReferencesToExpressions()
 					// column, so fall back on column name, which is unique
 					// then.
 					ftRC = rcl.getResultColumn(columnName);
+					tableNumber = ft.getTableNumber();
+				}
+				else {
+					tableNumber = ftRC.getTableNumber();
 				}
```

Comment:
Love the tests Jun, anyway you can add a comment here explaining the issue?  Maybe link to JIRA?